app/models/user.rb

```diff
@@ -66,7 +68,11 @@ def multiple_identities?
   end
 
   def active_profile
-    profiles.where(active: true).first
+    profiles.find(&:active?)
+  end
+
+  def verified?
+    active_profile ? true : false
   end
```

Comment:
agreed about `active_profile.present?`

About the UserDecorator, I will confess I am confused. IIUC decorator classes are for extending the Model for presentation purposes (View). The `verified?` method is supposed to indicate the state of the user for purposes of enforcing business logic. It isn't used for presentation at all.

Every other method in `UserDecorator` seems to be about presentation, except for `identity_not_verified?` -- I would argue that that method belongs in the Model because (again) it indicates model state, not anything to do with presentation.

Am I misunderstanding the decorator pattern?